src/IRTS/CodegenC.hs

```diff
@@ -179,7 +179,7 @@ bcc i (ASSIGNCONST l c)
     = indent i ++ creg l ++ " = " ++ mkConst c ++ ";\n"
   where
     mkConst (I i) = "MKINT(" ++ show i ++ ")"
-    mkConst (BI i) | i < (2^30) = "MKINT(" ++ show i ++ ")"
+    mkConst (BI i) | i > ((-2)^30) && i < (2^30) = "MKINT(" ++ show i ++ ")"
```

Comment:
@asandroq @melted Isn't `(-2)^30 == 2^30`?  So this condition is never true, and the code generator will now *always* generate a `MKBIGC`.  It should probably say `-(2^30)` instead?  (As an aside, why is it `2^30` rather than `2^31`?)

Reply:
Well, this is embarrassing. I am going to make another PR to fix this one.

Reply:
I pushed a fix in another PR, I hope I got this right this time. The reason for using 2^30 is that the last bit of a C int is used to tag it so it can be unboxed in the runtime.